src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,292 @@
+    };
+
+    const emitContentChanged = () => {
+        onTextChange && onTextChange(editorState);
+    };
+
+    const onLinkChooserConfirm = () => {
+        setSavedSelection(null);
+        setShowInlineToolbar(false);
+        setEditorArea(RichTextEditorArea.Normal);
+        emitContentChanged();
+    };
+
+    const onEditorBlur = () => {
+        setSavedSelection(null);
+        setShowInlineToolbar(false);
+        setEditorArea(RichTextEditorArea.Normal);
+        editor.current?.blur();
+    };
+
+    const onEditorClick = () => {
+        !showInlineToolbar && editor.current?.focus();
+    };
+
+    return (
+        <div onClick={onEditorClick}>
+            <Editor
+                ref={editor}
+                customStyleMap={styleMap}
+                editorState={editorState}
+                placeholder={placeholder}
+                onChange={onEditorChange}
+                onBlur={onEditorBlur}
+                handleKeyCommand={handleKeyCommand}
+                keyBindingFn={keyBindingFn}
+                readOnly={readonly}
+            />
+
+            <div ref={selectionRectRef} className="tw-absolute tw-pointer-events-none"></div>
+
+            <div
+                ref={inlineToolbarRef}
+                className={`arcade-toolbar tw-z-10 tw-drop-shadow-md ${
+                    !showInlineToolbar ? "tw-invisible tw-pointer-events-none" : ""
+                }`}
+                style={styles.popper}
+                {...attributes.popper}
+            >
+                <InlineToolbar
+                    onClick={emitContentChanged}
+                    store={{
+                        editorState,
+                        setEditorState,
+                        savedSelection,
+                    }}
+                >
+                    {(externalProps: InlineToolbarChildrenProps) => (
+                        <>
+                            {editorArea === RichTextEditorArea.Normal && (
+                                <>
+                                    <H3Button {...externalProps} />
+                                    <H4Button {...externalProps} />
+                                    <BoldButton {...externalProps} />
+                                    <ItalicButton {...externalProps} />
+                                    <UnderlineButton {...externalProps} />
+                                    <StrikethroughButton {...externalProps} />
+                                    <CodeButton {...externalProps} />
+                                    <SubButton {...externalProps} />
+                                    <SupButton {...externalProps} />
+                                    <OrderedListButton {...externalProps} />
+                                    <UnorderedListButton {...externalProps} />
+                                    <BlockquoteButton {...externalProps} />
+                                    <LinkButton onClick={onShowLinkChooser}></LinkButton>
```

Comment:
When a link is set, only the link preview is shown and the text editor cannot be accessed anymore